traveler-swift-core/TravelerKit/Models/Product.swift

```diff
@@ -42,10 +41,51 @@ struct AnyProduct: Decodable {
         case .booking:
             let bookingProduct = try BookingProduct(from: decoder)
             self.bookingProduct = bookingProduct
+            self.parkingProduct = nil
         case .parking:
-            throw DecodingError.dataCorruptedError(forKey: CodingKeys.productType , in: container, debugDescription: "Unsupported product type")
+            let parkingProduct = try ParkingProduct(from: decoder)
+            self.parkingProduct = parkingProduct
+            self.bookingProduct = nil
+        }
+    }
+}
+
+///Any purchased parking product:
+public struct ParkingProduct: Product, Decodable {
```

Comment:
Suggestion: Move each struct into its own file and encompass all into a `group` for better organization.

Reply:
Added a todo comment. for the moment we're leaving those models there because we're considering doing a bit of refactoring after IMR to distinguish between items in the catalog and items that have already been purchased. Naming convention and potential protocols for these models have not been considered or thought of so in order to avoid messing too much with the codebase I'm leaving those there for the refactoring.